gateway/README.md

```diff
@@ -11,149 +11,260 @@ This basic example demonstrates how to expose a service or a function through an
-5. Click the **Save** button.
+1. If you **didn't** follow the steps in **Expose a service without authentication** or **Expose a service with JWT authentication** section, go straight to step 2 of this instruction. If you did, you must delete the previously created APIRule. Select the **API Rules** button, click on the trash can icon next to the APIRule and confirm.
+2. Select the **Services** button and click on the name of the service you created. The name should be the same as the service name in the `deployment.yaml` file - **http-db-service**.
+3. In the **API Rules for http-db-service** section, click the **Expose API** button.
+4. Fill the **Name** text box.
+5. Fill the **Hostname** text box. The name you entered is referred to as the **\{hostname\}**. The domain next to it is referred to as the **\{domain\}**.
+6. In the **Access strategies** section, click on the dropdown and select the **OAuth2** field. 
+7. Fill the **Required scope** text box with `read, write`.
+8. Click the **Create** button.
 
-#### Fetch token
+#### Fetch OAuth2 token
 
-1. In the **Exposed APIs** section, click on the API you exposed.
-2. In the **Security** section, click the **Fetch token** button.
-3. Select the whole text and copy it to the clipboard, or click the **Copy to clipboard** button. Make sure that the word `Bearer` is also copied.
-4. The token is later referred to as **\{token\}**.
+1. On the Namespace main page, click **Deploy new resource** button, select the `oauth2client.yaml` file from the `service` directory in this example, and click **Upload**.
+2. Fetch the access token with required scopes. The access token in the response is later referred to as **\{oauth2-token\}**. Run:
 
-#### Test the APIs with authentication
+    ```bash
+    curl https://oauth2.{domain}/oauth2/token -H "Authorization: Basic ZXhhbXBsZS1pZDpleGFtcGxlLXNlY3JldA==" -F "grant_type=client_credentials" -F "scope=read write"
+    ```
+
+#### Test the APIs with OAuth2 authentication
 
 ```bash
 # To perform a test without the token, use the following command:
-curl -ik https://{hostname}.kyma.local
-# > 401 Origin authentication failed.
+curl https://{hostname}.{domain}/orders
+# > {"error":{"code":401,"status":"Unauthorized","request":"1915853b-9780-4751-b26d-903a179e2941","message":"The request could not be authorized"}}
 
 # To perform a test with the token, use the following command:
-curl -ik https://{hostname}.kyma.local -H 'Authorization: {token}'
-# > 200 Hello world
+curl -ik https://{hostname}.{domain}/orders -H 'Authorization: Bearer {oauth2-token}'
+# > 200 []
 ```
+ 
+>**NOTE:** If you are using the Kyma deployed locally, add the `{hostname}.kyma.local` to your hosts file.
 
 ### Manual exposure using kubectl
 
-There are additional prerequisites to exposing a service or a function manually using kubectl:
+There are additional prerequisites to exposing a service manually using kubectl:
```

Comment:
I would suggest adding an additional prerequisite here. The user has to have kubectl configured and pointing to the proper cluster before they start following the steps below. IMHO we should not assume, that they know how to do that, or that it is automatically done. I suggest adding a quick `configuration` guide to check if the kubeconfig is configured (`kubectl cluster-info`) and if not then fetch the kubeconfig from the Console UI and export it.

Reply:
I'm not sure if a guide on how to fetch Kyma kubeconfig should be a part of this example. I will add a link to the guide in our documentation, though.

Reply:
You are right. A link to the already existing guide in our documentation is a great idea